### PR TITLE
Osmm/speed upgrade

### PR DESCRIPTION
**Pull request by ykarmesh**
## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py [hunk 1]
```diff
@@ -143,13 +145,31 @@ def _try_acquire_context(self):
...
 
+    def _auto_sleep(self):
+        all_robo_pos = [
+            robot.base_pos for robot in self.robots_mgr.robots_iter
+        ]
+        rom = self.get_rigid_object_manager()
+        for handle, ro in rom.get_objects_by_handle_substring().items():
+            is_far = all(
+                (robo_pos - ro.translation).length()
+                > self.habitat_config.sleep_dist
```
Comment (yvsriram) [inline habitat-lab/habitat/tasks/rearrange/rearrange_sim.py:156]: Accessing hydra config. We will need to change similar to the other [PR](https://github.com/facebookresearch/habitat-lab/pull/1137/). I will remember to update this.

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py [hunk 2]
```diff
@@ -668,6 +694,8 @@ def step(self, action: Union[str, int]) -> Observations:
             self.viz_ids = defaultdict(lambda: None)
 
         self.maybe_update_robot()
+        if self.habitat_config.sleep_dist > 0.0:
```
Comment (yvsriram) [inline habitat-lab/habitat/tasks/rearrange/rearrange_sim.py:697]: Same

**Review by yvsriram — COMMENTED** [review]
Do we want to train and test these changes out before merging? @ASzot

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py (hunk 1 above)
Comment (ykarmesh) [reply habitat-lab/habitat/tasks/rearrange/rearrange_sim.py:156]: Thanks Sriram!

**Review by yvsriram — APPROVED** [review]

**Pull request merged**